scanner/src/main/kotlin/experimental/ScannerWrapper.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.experimental
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.KnownProvenance
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScanSummary
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+import org.ossreviewtoolkit.scanner.ScanResultsStorage
+import org.ossreviewtoolkit.scanner.ScannerCriteria
+
+/**
+ * An interface to wrap source code scanners.
+ */
+sealed interface ScannerWrapper {
+    /**
+     * The name of the scanner.
+     */
+    val name: String
+
+    /**
+     * The details of the scanner.
+     */
+    val details: ScannerDetails
+
+    /**
+     * Return a [ScannerCriteria] object to be used when looking up existing scan results from a [ScanResultsStorage].
+     * By default the properties of this object are initialized by the scanner implementation. These default can be
+     * overridden with the [ScannerConfiguration.options] property: Use properties of the form
+     * _scannerName.criteria.property_, where _scannerName_ is the name of the scanner and _property_ is the name of a
+     * property of the [ScannerCriteria] class. For instance, to specify that a specific minimum version of ScanCode is
+     * allowed, set this property: `options.ScanCode.criteria.minScannerVersion=3.0.2`.
+     */
+    fun getScannerCriteria(): ScannerCriteria
```

Comment:
I'm probably missing something, but why is this not also a property, like the ones above?

Reply:
Mainly because the existing implementations already have this function, I thought that's something to better clean up once we reach a state were the old implementation can be removed.

Reply:
I believe the new scanner implementation is so different already anyway that we don't have to be considerate of eventual limitations of the existing implementations. My fear simply is that we forget about this if we don't do it now.

Reply:
I have made it a property.